
version: '3'

services:
  webServer:
    image: nginx:1.17.10
    container_name: nginx_container
    depends_on:
        - api
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    ports:
      - 80:80

  api:
    build: .
    container_name: api_container
    depends_on:
      - db
    environment:
      - POSTGRES_NAME=$POSTGRES_NAME
      - POSTGRES_USER=$POSTGRES_USER
      - POSTGRES_PASSWORD=$POSTGRES_PASSWORD
      - POSTGRES_HOST=$POSTGRES_HOST
      - POSTGRES_PORT=$POSTGRES_PORT
      - ALLOWED_HOSTS=$ALLOWED_HOSTS
    command: python backend/manage.py runserver 0.0.0.0:5001
    ports: 
      - "5001:5001"
    restart: on-failure

  db:
    image: postgres
    volumes:
      - ./data/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=$POSTGRES_DB
      - POSTGRES_USER=$POSTGRES_USER
      - POSTGRES_PASSWORD=$POSTGRES_PASSWORD
    container_name: db_container
    healthcheck:
      test: [ "CMD", "pg_isready", "-q", "-d", "postgres", "-U", "root" ]
      timeout: 45s
      interval: 10s
      retries: 10
    restart: always
    ports:
      - '5432:5432'